src/System.Net.HttpListener/tests/HttpListenerFactory.cs

```diff
@@ -55,6 +56,17 @@ internal HttpListenerFactory()
                     // in trying again.
                     if (!(e is HttpListenerException) && !(e is SocketException))
                         break;
+
+                    // If we can't access the host (e.g. if it is '+' or '*' and the current user is the administrator)
+                    // then throw.
+                    if (e is HttpListenerException listenerException)
+                    {
+                        const int ERROR_ACCESS_DENIED = 5;
+                        if (listenerException.ErrorCode == ERROR_ACCESS_DENIED && (hostname == "*" || hostname == "+"))
+                        {
+                            throw new InvalidOperationException($"Access denied for host {hostname}");
+                        }
+                    }
```

Comment:
Yeah that's cleaner - done. I'm not sure if the SocketException stuff is dead code as that looks like something only the managed implementation would through (these situations have been fixed) - but no harm leaving it in for now.